specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01/redis.json

```diff
@@ -1850,85 +1698,583 @@
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "Redis",
+          "AAD"
+        ],
+        "operationId": "AccessPolicyAssignment_Get",
+        "x-ms-examples": {
+          "RedisCacheAccessPolicyAssignmentGet": {
+            "$ref": "./examples/RedisCacheAccessPolicyAssignmentGet.json"
+          }
+        },
+        "description": "Gets the list of assignments for an access policy of a redis cache",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "cacheName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Redis cache.",
+            "pattern": "^([a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]|[a-zA-Z0-9])$"
+          },
+          {
+            "name": "accessPolicyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the assigned access policy.",
+            "pattern": "^[a-zA-Z0-9]+$"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Response of get access policy assignments.",
+            "schema": {
+              "$ref": "#/definitions/RedisCacheAccessPolicyAssignmentRequestResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/redis/{cacheName}/accessPolicyAssignments": {
+      "get": {
+        "tags": [
+          "Redis",
+          "AAD"
+        ],
+        "operationId": "AccessPolicyAssignment_List",
+        "x-ms-examples": {
+          "RedisCacheAccessPolicyAssignmentList": {
+            "$ref": "./examples/RedisCacheAccessPolicyAssignmentList.json"
+          }
+        },
+        "description": "Gets the list of access policy assignments associated with this redis cache",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "cacheName",
```

Comment:
why don't use reference like:
https://github.com/Azure/azure-rest-api-specs/blob/813022a095cd9cfd919795167689053a286890d4/specification/redis/resource-manager/Microsoft.Cache/preview/2023-05-01/redis.json#L1623-L1625

Reply:
I think we need to refactor to use one parameter.

```
"CacheNameParameter": {      
    "name": "cacheName",      
    "in": "path",     
     "required": true,      
    "type": "string",      
    "description": "The name of the Redis cache.",      
    "x-ms-parameter-location": "method",      
    "pattern": "^([a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]|[a-zA-Z0-9])$"   
 }

```

Reply:
Referred to cacheName parameter everywhere now